Spurious warning: Cache is deprecrated

Using Xcode 6.3.2, compiling Bugsnag (4.0.7) shows the following warning:

```
In file included from /Users/.../Pods/Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/KSObjC.c:29:
/Users/.../Pods/Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/KSObjCApple.h:122:5: warning: 'Cache' is deprecated: first deprecated in iOS 2.0 [-Wdeprecated-declarations]
    Cache cache;
    ^
/Users/.../Pods/Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/KSObjCApple.h:51:28: note: 'Cache' has been explicitly marked deprecated here
typedef struct objc_cache *Cache;
                           ^
1 warning generated.
```

This doesn't seem to be a problem. An easy fix would be to add the a #pragma to disable the warning.

--- a/KSCrash/Source/KSCrash/Recording/Tools/KSObjCApple.h
+++ b/KSCrash/Source/KSCrash/Recording/Tools/KSObjCApple.h
@@ -119,7 +119,10 @@ typedef struct class_rw_t {
 typedef struct class_t {
     struct class_t *isa;
     struct class_t *superclass;
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
     Cache cache;
+#pragma clang diagnostic pop
     IMP *vtable;
     uintptr_t data_NEVER_USE;  // class_rw_t * plus custom rr/alloc flags
 } class_t;

